src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -522,6 +522,40 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 			break;
 		}
 
+		case T_CreateStatsStmt:
+		{
+			CreateStatsStmt *createStatsStmt = (CreateStatsStmt *) parseTree;
+
+			/* because CREATE STATISTICS statements can only have one relation */
+			RangeVar *relation = linitial(createStatsStmt->relations);
+
+			char **relationName = &(relation->relname);
+			char **objectSchemaName = &(relation->schemaname);
+
+			SetSchemaNameIfNotExist(objectSchemaName, schemaName);
+			AppendShardIdToName(relationName, shardId);
+
+			Value *statNameValue = llast(createStatsStmt->defnames);
+			char *statName = strVal(statNameValue);
+			char **statNameVar = &(statName);
+			AppendShardIdToName(statNameVar, shardId);
+			statNameValue = makeString(pstrdup(statName));
```

Comment:
this seems very complex code for what we want to do.
currently it finds the original pointer to where we store the string pointer so to inplace append the shardid.
then we copy the value out of it to create our new `NameList`.

Can we instead copy the `strVal` somewhere local and pass the address of that into `AppendShardIdToName`. This will not destroy or alter the original memory while also making it easier to follow.

```suggestion
			Value *statNameValue = llast(createStatsStmt->defnames);
			char *statName = strVal(statNameValue);
			statName = pstrdup(statName)
			AppendShardIdToName(&statName, shardId);
```